core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -393,12 +393,20 @@ private[deploy] class Worker(
 
   private def handleRegisterResponse(msg: RegisterWorkerResponse): Unit = synchronized {
     msg match {
-      case RegisteredWorker(masterRef, masterWebUiUrl, masterAddress) =>
-        if (preferConfiguredMasterAddress) {
-          logInfo("Successfully registered with master " + masterAddress.toSparkURL)
-        } else {
-          logInfo("Successfully registered with master " + masterRef.address.toSparkURL)
+      case RegisteredWorker(masterRef, masterWebUiUrl, masterAddress, duplicate) =>
+        val preferredMasterAddress = {
+          if (preferConfiguredMasterAddress) {
+            masterAddress.toSparkURL
+          } else {
+            masterRef.address.toSparkURL
+          }
+        }
+
+        if (duplicate) {
+          logWarning(s"Duplicate registration at master $preferredMasterAddress")
```

Comment:
would be good to include the possible reasons for this case, so that end users know what's going on.

Reply:
Good idea, I'll update it.